Tests/Functional/Drivers/Queue/PhpAmqpLibDriverTest.php

```diff
@@ -0,0 +1,259 @@
+        $this->driver->consume($this->consumer->getName(), $this->queueName);
+        $this->driver->destroy($this->consumer->getName());
+        $this->assertFalse($this->driver->isConnected());
+    }
+
+     /**
+     * Tests that tries to declare a channel without having a connection
+     */
+    public function testDeclareChannelWithoutConnnection()
+    {
+        $this->expectException(AMQPConnectionClosedException::class);
+        $this->expectExceptionMessage('Broken pipe or closed connection');
+        $this->assertTrue($this->driver->isConnected());
+        $this->driver->declareChannel();
+        $this->driver->destroy($this->createConsumer()->getName());
+        $this->assertFalse($this->driver->isConnected());
+        $this->driver->declareChannel();
+    }
+
+    /**
+     * Test the format parameter set in the class
+     */
+    public function testFormat()
+    {
+        $format = 'text/plain';
+        $this->driver->setFormat($format);
+        $this->assertEquals($format, $this->driver->getFormat());
+    }
+
+    /**
+     * @throws AMQPProtocolException
+     */
+    public function testConnectWithoutData()
+    {
+        $this->driver->declareChannel();
+        $consumer = $this->createConsumer();
+        $this->driver->destroy($consumer->getName());
+        $this->driver->configure('', '', '', '');
+        $this->driver->connect();
+        $this->assertTrue($this->driver->isConnected());
+        $this->assertInstanceOf(AMQPStreamConnection::class, $this->driver->declareChannel()->getConnection());
+    }
+
+    /**
+     * @dataProvider getMessages
+     */
+    public function testConsumeWaitNoBlock(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+        $amqpMessage = null;
+
+        $callback = function($message) use (&$amqpMessage){
+            $amqpMessage = $message;
+            $queueMessage = new QueueMessage();
+            $queueMessage->setMessageId($message->getDeliveryTag());
+            $this->driver->ack($queueMessage);
+        };
+
+        $this->driver->consume($this->consumer->getName(), $this->queueName, $callback);
+        $this->driver->declareChannel()->is_consuming();
+        $this->driver->waitNoBlock();
+
+        $this->assertInstanceOf(AMQPMessage::class, $amqpMessage);
+        $this->assertEquals($amqpMessage->delivery_info['routing_key'], $this->queueName);
+        $this->assertEquals($amqpMessage->delivery_info['consumer_tag'], $this->consumer->getName());
+        $this->assertNotNull($amqpMessage->getBody());
+    }
+
+    /**
+     * Prepare the data and class to consume a message
+     *
+     * @param MessageInterface $msg
+     * @return mixed
+     */
+    public function prepareToConsume(MessageInterface $msg)
+    {
+        $msgIn = $this->createQueueMessage($msg);
+        $msgIn->addHeader('test_header', '12345');
+        $this->driver->send($msgIn);
+        $this->driver->disconnect();
+        $this->driver->connect();
```

Comment:
Which is the purpose of sending the message, disconnecting and connecting again?

Reply:
To emulate different process (publish/consume). So, in this way we have separate and dedicated process for each one.

Reply:
But here there is no consume, just publish a message